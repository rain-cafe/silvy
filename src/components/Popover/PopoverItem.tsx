import styles from './PopoverItem.module.scss';

export interface PopoverItemProps {
    children: React.ReactNode;
}

export function PopoverItem({
    children
}: PopoverItemProps) {
    return (
        <div className={styles.popoverItem}>
            {children}
        </div>
    );
}