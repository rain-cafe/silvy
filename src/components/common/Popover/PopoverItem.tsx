import { IconType } from 'react-icons';
import { PolymorphicPropsWithoutRef } from 'react-polymorphic-types';
import styles from './PopoverItem.module.scss';

const DEFAULT_ELEMENT = 'button';

export interface PopoverItemOwnProps {
    children: React.ReactNode;
    icon?: IconType;
}

export type PopoverItemProps<
    T extends React.ElementType = typeof DEFAULT_ELEMENT
> = PolymorphicPropsWithoutRef<PopoverItemOwnProps, T>;

export function PopoverItem<E extends React.ElementType = typeof DEFAULT_ELEMENT>({
    as,
    children,
    icon: Icon,
    ...extraProps
}: PopoverItemProps<E>) {
    // eslint-disable-next-line @typescript-eslint/naming-convention
    const Element: React.ElementType = as || DEFAULT_ELEMENT;

    return (
        <Element className={styles.popoverItem} {...extraProps}>
            {Icon && (
                <Icon size={20}/>
            )}
            <div className={styles.content}>
                {children}
            </div>
        </Element>
    );
}