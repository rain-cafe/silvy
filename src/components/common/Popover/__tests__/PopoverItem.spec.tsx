import { createRenderComponent } from '@rain/test-utils/render';
import { PopoverItem, PopoverItemProps } from '../PopoverItem';

describe('Component(PopoverItem)', () => {
    const renderPopoverItem = createRenderComponent<PopoverItemProps>(PopoverItem);

    it('should display the PopoverItem', () => {
        const { baseElement } = renderPopoverItem()

        expect(baseElement).toMatchSnapshot()
    });

    it('should support overriding the type', () => {
        const { baseElement } = renderPopoverItem<PopoverItemProps<'a'>>({
            as: 'a',
            href: 'https://silvy.rains.cafe'
        })

        expect(baseElement).toMatchSnapshot()
    });
});
