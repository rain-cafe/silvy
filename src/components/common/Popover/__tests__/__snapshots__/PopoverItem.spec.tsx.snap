// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(PopoverItem) should display the PopoverItem 1`] = `
<body>
  <div>
    <button
      class="popoverItem"
    >
      <div
        class="content"
      />
    </button>
  </div>
</body>
`;

exports[`Component(PopoverItem) should support overriding the type 1`] = `
<body>
  <div>
    <a
      class="popoverItem"
      href="https://silvy.rains.cafe"
    >
      <div
        class="content"
      />
    </a>
  </div>
</body>
`;
